### PR TITLE
Fixed Issue: #1977

### PR DESCRIPTION
**Pull request by SSivakumar12**
First of all, I want to take the opportunity to thank the awesome work being done by Martin and all contributors both past and present. This is a really good tool that we have used at work and has provided us with a lot of meaningful insights so I figured that I could do my bit to contribute to a tool that has helped me previously :)

This is my first PR and I sincerely hope it is not my last  especially within this repo since I really want to give back to this project. 

# What does this PR do?
Address the issue raised in issues #1977 which I believe has not been resolved yet.

Fixes # (issue)
I have added an edge case so that if a tokenizer isn't any of the expected patterns a ValueError is raised and produced a constructive error message to hopefully to prevent a repeat of the error. I did not update the documentation and add any tests. That being said, I am more than happy to add tests if deemed appropriate/valuable

## Before submitting
- [ ] This PR fixes a typo or improves the docs (if yes, ignore all other checks!).
- [x] Did you read the [contributor guideline](https://github.com/MaartenGr/BERTopic/blob/master/CONTRIBUTING.md)?
- [x] Was this discussed/approved via a Github issue? Please add a link to it if that's the case.
- [x] Did you make sure to update the documentation with your changes (if applicable)?
      - didn't update documentation but added a error message
- [x] Did you write any new necessary tests?
      - no but happy to do so in a separate PR since I think it doesn't fall into the scope of this current PR?

### CODE REVIEW

**Comment by MaartenGr** [comment]
Thank you for the PR! Is this exception raised whenever the documents are actually truncated or when you instantiate the related LLM? If it is the former, you would get an error after the clustering has been done which seems like wasted resources for the user. Would it perhaps be an idea to raise this error earlier?

**Comment by SSivakumar12** [comment]
Good point! In my case right now, the error would be raised when I instantiate the relevant LLM. I will look to try and find a suitable place to raise the earlier and amend accordingly.

**Comment by SSivakumar12** [comment]
Looking at an example implementation for an representation class such as [llamacpp](https://github.com/MaartenGr/BERTopic/blob/master/bertopic/representation/_llamacpp.py#L148),  the purpose of `truncate_document`, is to prepare the prompt so that it could be sent to the LLM?

I might be wrong so I do apologise but wouldn't this mean you wouldn't get an error after clustering? Therefore where it is reasonable?

File: bertopic/representation/_cohere.py [hunk 1]
```diff
@@ -126,12 +126,7 @@ def __init__(
         self.tokenizer = tokenizer
         self.prompts_ = []
 
-        if self.tokenizer is None and self.doc_length is not None:
-            raise ValueError(
-                "Please select from one of the valid options for the `tokenizer` parameter: \n"
-                "{'char', 'whitespace', 'vectorizer'} \n"
-                "If `tokenizer` is of type callable ensure it has methods to encode and decode a document \n"
-            )
+        _ = validate_truncate_document_parameters(self.tokenizer, self.doc_length)
```
Comment (MaartenGr) [inline bertopic/representation/_cohere.py:129]: Are you returning the `_` for a specific reason? Looking at the `validate_truncate_document_parameters`, there doesn't seem anything that is returned.

File: bertopic/representation/_utils.py [hunk 2]
```diff
@@ -58,6 +58,15 @@ def decode(self, doc_chunks):
...
     return document
 
+def validate_truncate_document_parameters(tokenizer, doc_length) -> Union[None, ValueError]:
+    """validates parameters that are used in the function `truncate_document`"""
+    if tokenizer is None and doc_length is not None:
+        raise ValueError(
+            "Please select from one of the valid options for the `tokenizer` parameter: \n"
+            "{'char', 'whitespace', 'vectorizer'} \n"
+            "If `tokenizer` is of type callable ensure it has methods to encode and decode a document \n"
+        )
```
Comment (MaartenGr) [inline bertopic/representation/_utils.py:68]: Should we also include a check for the opposite? That someone does use a tokenizer but not a doc_length?

File: bertopic/representation/_utils.py [hunk 3]
```diff
@@ -57,6 +58,21 @@ def decode(self, doc_chunks):
...
+    elif tokenizer is not None and doc_length is None:
+        raise ValueError(
+            "If `tokenizer` is provided, `doc_length` of type int must be provided as well."
+        )
+    else:
+        pass
```
Comment (MaartenGr) [inline bertopic/representation/_utils.py:74]: I believe this isn't necessary right?

Comment (SSivakumar12) [reply bertopic/representation/_utils.py:74]: No it is not necessary but I included it to be verbose in terms of the possible edgecase. That being said, happy to remove it.

Comment (MaartenGr) [reply bertopic/representation/_utils.py:74]: Let's remove it and I'll make sure to merge it after that! Other than that, it looks great!

**Comment by MaartenGr** [comment]
@SSivakumar12 Could you check the linting? It is failing there.

**Comment by SSivakumar12** [comment]
Apologies,  for the oversight, have resolved the linting issues locally and the build should pass on linting now

**Comment by MaartenGr** [comment]
Awesome, everything is looking good. Thank you for your work on this!

**Pull request merged**